src/Web/Sfa.Das.Sas.ApplicationServices/Handlers/SitemapHandler.cs

```diff
@@ -54,21 +53,56 @@ public SitemapResponse Handle(SitemapQuery message)
             };
         }
 
+        private IEnumerable<string> GetFrameworkDetailsInSeoFormat()
+        {
+            var frameworks = _getFrameworks.GetAllFrameworks();
+
+            return BuildFrameworkSitemap(frameworks);
+        }
+
+        private IEnumerable<string> GetStandardDetailsInSeoFormat()
+        {
+            var standards = _getStandards.GetAllStandards().Where(s => s.IsPublished);
+
+            return BuildStandardSitemap(standards);
+        }
+
         private IEnumerable<string> GetProviderDetailsInSeoFormat()
         {
             var providersExcludingEmployerProviders = _getProviders.GetAllProviders().Where(x => x.IsEmployerProvider == false);
-            var identifiers = BuildProviderSitemapFromProviders(providersExcludingEmployerProviders);
-            return identifiers;
+
+            return BuildProviderSitemapFromProviders(providersExcludingEmployerProviders);
+        }
+
+        private IEnumerable<string> BuildFrameworkSitemap(IEnumerable<Framework> frameworks)
+        {
+            return from framework in frameworks
+                   let title = EncodeTitle(framework)
+                   select GetSeoFormat(framework.FrameworkId, title);
+        }
+
+        private IEnumerable<string> BuildStandardSitemap(IEnumerable<Standard> standards)
+        {
+            return from standard in standards
+                   let title = EncodeTitle(standard)
+                   select GetSeoFormat(standard.StandardId, title);
         }
 
         private IEnumerable<string> BuildProviderSitemapFromProviders(IEnumerable<ProviderSummary> providers)
         {
-            foreach (var provider in providers)
-            {
-                var encodedProviderName = _urlEncoder.EncodeTextForUri(provider.ProviderName);
-                var urlLocElement = $@"{provider.Ukprn}/{encodedProviderName}";
-                yield return urlLocElement;
-            }
+            return from provider in providers
```

Comment:
could you revert this to use the foreach just for consistency across the code base. also changing this to a select could break any upstream enumerating.

likewise the two methods above

Reply:
I've reverted these back to foreach